lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php

```diff
@@ -703,6 +703,13 @@ public function getMigrationsToExecute($direction, $to)
         return $versions;
     }
 
+
+    public function setMappingTypes($mapping_types) {
+        foreach($mapping_types as $field=>$type) {
+            $this->connection->getDatabasePlatform()->registerDoctrineTypeMapping($field, $type);
```

Comment:
changing the configuration of the connection looks wrong to me. the connection is passed as a constructor argument to this class, which means that configuring it is meant to be done outside this class